zerver/worker/queue_processors.py

```diff
@@ -607,17 +609,33 @@ def maybe_send_batched_emails(self) -> None:
 
 
 @assign_queue("email_senders")
-class EmailSendingWorker(QueueProcessingWorker):
+class EmailSendingWorker(LoopQueueProcessingWorker):
+    def __init__(self) -> None:
+        super().__init__()
+        self.connection: EmailBackend = initialize_connection(None)
+
     @retry_send_email_failures
-    def consume(self, event: Dict[str, Any]) -> None:
+    def send_email(self, event: Dict[str, Any]) -> None:
+        send_email_from_dict(event)
+
+    def consume_batch(self, events: List[Dict[str, Any]]) -> None:
         # Copy the event, so that we don't pass the `failed_tries'
-        # data to send_email_from_dict (which neither takes that
-        # argument nor needs that data).
-        copied_event = copy.deepcopy(event)
-        if "failed_tries" in copied_event:
-            del copied_event["failed_tries"]
-        handle_send_email_format_changes(copied_event)
-        send_email_from_dict(copied_event)
+        # data to send_email (which neither takes that argument nor
+        # needs that data).
+        copied_events = copy.deepcopy(events)
+        for copied_event in copied_events:
+            if "failed_tries" in copied_event:
+                del copied_event["failed_tries"]
+            copied_event["connection"] = self.connection
```

Comment:
Before we pass the connection down, we should make sure that it's "fresh."
```suggestion
            self.connection = initialize_connection(self.connection)
            copied_event["connection"] = self.connection
```